apps/overige-objecten-api/src/utils/processCsvFile.ts

```diff
@@ -0,0 +1,54 @@
+import csvParser from 'csv-parser';
+import fs from 'node:fs';
+import { v4 } from 'uuid';
+import { sanitizeHTML } from './sanitizeHTML';
+
+export const processCsvFile = (filePath: string, requiredColumns: string[]) => {
+  const results: Record<string, string>[] = [];
+  let hasRequiredColumns = true;
+
+  // Create read stream for CSV file
+  return new Promise<any[]>((resolve, reject) => {
+    fs.createReadStream(filePath)
+      .pipe(csvParser())
+      .on('headers', (headers) => {
+        // Check if required columns exist in the CSV file headers
+        const missingColumns = requiredColumns.filter((col) => !headers.includes(col));
+
+        if (missingColumns.length > 0) {
+          // If any required columns are missing, reject with error
+          hasRequiredColumns = false;
+          reject({ error: 'Missing required columns', missingColumns });
+        }
+      })
+      .on('data', (data) => {
+        // If the columns are valid, push the data into results
+        if (hasRequiredColumns) {
+          results.push({ vraag: data.vraag, antwoord: data.antwoord });
+        }
+      })
+      .on('end', () => {
+        if (hasRequiredColumns) {
+          const sanitizedResults = results.map((result) => {
+            const DOMPurifyHTML = sanitizeHTML(result?.antwoord);
```

Comment:
Je gebruikt nu "constructor case" voor een variable. Ik stel voor om `domPurifyHTML` te gebruiken, om duidelijk te maken dat het geen constructor is maar een waarde.